apis/python/src/tiledbsc/soma_collection.py

```diff
@@ -125,3 +134,124 @@ def cell_count(self) -> int:
         Returns sum of `soma.cell_count()` over SOMAs in the collection.
         """
         return sum(soma.cell_count() for soma in self)
+
+    # ----------------------------------------------------------------
+    def attribute_filter(
```

Comment:
couple of issues:
* the requirement to redundantly provide the attribute names (ie, both in the query string and as a separate attr_name) seems unnecessary.  Other than the early error check, what is the purpose of `obs_attr_names` and `var_attr_names`
* this routine returns only obs, var and X - what about the other obs/var indexed sub-objects in the collection, such as raw, obsm, etc?